### PR TITLE
volgen: Add support for Thin Arbiter Volfile

### PR DESCRIPTION
**Pull request by aravindavk**
Updates: #655
Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

**Comment by prashanthpai** [comment]
**cc* @aspandey

File: glusterd2/volgen2/clustergraph.go [hunk 1]
```diff
@@ -72,6 +72,32 @@ func clusterGraph(volfile *Volfile, dht *Entry, vol *volume.Volinfo, nodeid uuid
 					decommissionedBricks = append(decommissionedBricks, name)
 				}
 			}
+
+			// FIXME: Change the name of thin arbiter option
+			thinarbiter, exists := vol.Options["cluster.replicate.thin-arbiter"]
+			if exists && thinarbiter != "" {
+				taParts := strings.Split(thinarbiter, ":")
```
Comment (amarts) [inline glusterd2/volgen2/clustergraph.go]: I would still like to keep it as 'host<:port>:/brick'  (whatever is the normal standard), so we can handle the case of different port of thin-arbiter from the day1 itself.
If its going to be difficult, it is ok, but assumption is most of the time user doesn't specify the port (in that case 24007 is fine), but if he specifies, we get control of handling this.

File: glusterd2/volgen2/clustergraph.go [hunk 2]
```diff
@@ -72,6 +72,32 @@ func clusterGraph(volfile *Volfile, dht *Entry, vol *volume.Volinfo, nodeid uuid
 					decommissionedBricks = append(decommissionedBricks, name)
 				}
 			}
+
+			// FIXME: Change the name of thin arbiter option
```
Comment (prashanthpai) [inline glusterd2/volgen2/clustergraph.go]: Both the `FIXME` needs to be handled before this can be merged.

**Comment by prashanthpai** [comment]
@aravindavk Ravi gave this a try today. He also needs the value set by user for the thin-arbiter option to end up in xlator options in the volfile. It should've ended up in volfile (as per normal volume set code) but it didn't.

**Comment by aravindavk** [comment]
Modified PR and addressed the review comments. Please review

**Pull request merged**